app/(auth)/register.tsx

```diff
@@ -122,6 +119,57 @@ const Register = () => {
           Sign up
         </Button>
       </Stack>
+
+      <Modal
```

Comment:
I think the app should not be aware of a "demo" mode, and only react to api errors